src/models/kv_cache.cpp

```diff
@@ -376,6 +376,33 @@ std::string ComposeKeyValueName(const std::string& template_string, int index) {
   return std::string(key_value_name);
 }
 
+ModelManagedKeyValueCache::ModelManagedKeyValueCache(State& state)
+    : state_{state} {
+  // This fixes with _qa sample where a new instance of ModelManagedKeyValueCache is
+  // created for each new prompt from user. In this case, we need to trigger a KVCache
+  // reset on the session before the first Session::Run.
```

Comment:
A comment explaining the coupling of the kv cache state with the ort session state and its implications would be helpful.

```suggestion
  // A new instance of ModelManagedKeyValueCache is created for each Generator.
  // In this case, we need to trigger a KVCache reset on the session before the first Session::Run.
  // This implies that the key-value cache state is coupled with the ONNX Runtime Session and
  // that only 1 Generator can be active for the Model at any given time.
```

In theory, the Model:Generator relationship is 1:many since the ort session is mostly stateless. But for openvino, the ort-session is stateful. Which means the Model:Generator relationship is 1:1 at any given time.

Reply:
Yes, correct. We are aware of this, and have some ideas about how to address it. Within the session, we can have multiple ov::InferRequest objects, each with their own states -- we just need a way to tie them to a particular instance of a Generator within ORT GenAI. Anyway, absolutely agree that supporting 1:many is needed later on.